### PR TITLE
Feature #1544 generate_numeric_field improvements

### PR DESCRIPTION
**Pull request by PaulBender**
Feature #1544

### CODE REVIEW

**Comment by ATofighi** [comment]
> We can’t automatically merge this pull request.

**Comment by PaulBender** [comment]
Fixed

File: admin/inc/class_form.php [hunk 1]
```diff
@@ -833,7 +833,7 @@ function generate_date_select($name, $day="",$month="",$year="")
 
 		$built = $this->generate_select_box($name.'_day', $days, (int)$day, array('id' => $name.'_day'))." &nbsp; ";
 		$built .= $this->generate_select_box($name.'_month', $months, (int)$month, array('id' => $name.'_month'))." &nbsp; ";
-		$built .= $this->generate_text_box($name.'_year', (int)$year, array('id' => $name.'_year', 'style' => 'width: 100px;'));
+		$built .= $this->generate_numeric_field($name.'_year', (int)$year, array('id' => $name.'_year', 'style' => 'width: 100px;', 'min' => 0));
```
Comment (ATofighi) [inline admin/inc/class_form.php]: I think `(int)` isn't necessary...

Comment (JN-Jones) [inline admin/inc/class_form.php]: The value is casted in `generate_numeric_field` so yes, it isn't necessary.

Comment (PaulBender) [inline admin/inc/class_form.php]: I've removed it

**Pull request merged**